ultravox/tools/ds_tool.py

```diff
@@ -138,9 +148,13 @@ def main(args: DatasetToolArgs):
 
     try:
         if args.dataset_split:
-            data_dict[args.dataset_split].push_to_hub(args.upload_name, **hub_args)
-        else:
-            data_dict.push_to_hub(args.upload_name, **hub_args)
+            # load the full dataset, otherwise the existing splits will be overwritten
+            upload_ds = datasets.load_dataset(args.upload_name, args.dataset_subset)
```

Comment:
should we just push the individual split in this case (not as a `DatasetDict`?)

Reply:
I'll take another look. I think that was what I was doing before and it ended up not working correctly, but I could be wrong.

Reply:
You're right. The previous version somehow had missed the `split=...` argument, so the fix was way simpler.